src/Credentials.js

```diff
@@ -145,21 +147,31 @@ class Credentials {
   *  Send a push notification to a user, consumes a token which allows you to send push notifications
   *  and a url/uri request you want to send to the user.
   *
-  *  @param    {String}                  token       a push notification token (get a pn token by requesting push permissions in a request)
+  *  @param    {String}                  token              a push notification token (get a pn token by requesting push permissions in a request)
+  *  @param    {String}                  pubEncKey          the public encryption key of the receiver, encoded as a base64 string
+  *  @param    {String}                  payload.url        a uport request url
+  *  @param    {String}                  payload.message    a message to display to the user
   *  @return   {Promise<Object, Error>}              a promise which resolves with successful status or rejects with an error
   */
-  push (token, {url}) {
+  push (token, pubEncKey, {url, message}) {
     return new Promise((resolve, reject) => {
       if (!token) {
         return reject(new Error('Missing push notification token'))
       }
       if (!url) {
         return reject(new Error('Missing payload url for sending to users device'))
       }
+      if (!pubEncKey) {
+        return reject(new Error('Missing public encryption key of the receiver'))
+      }
+
+      const plaintext = padMessage(JSON.stringify({url, message}))
+
+      const enc = encryptMessage(plaintext, pubEncKey)
 
       nets({
-        uri: 'https://pututu.uport.me/api/v1/sns',
-        json: {url},
+        uri: 'https://pututu.uport.space/api/v2/sns', // TODO - change to .me
+        json: { message: JSON.stringify(enc) },
```

Comment:
assuming mobile will support both encrypted and unencrypted for a while?

Reply:
yup

Reply:
👍